# Importing the pandas ,seaborn and matplotlib,stats librararies¶

In [14]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stat

# Reading the Sales dataset

In [3]:
data=pd.read_csv("Sales_add.csv")
data.head(5)

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1008.0+ bytes


# Case Study on Testing of Hypothesis


A company started to invest in digital marketing as a new way of their product
promotions.For that they collected data and decided to carry out a study on it.
● The company wishes to clarify whether there is any increase in sales after
stepping into digital marketing.
● The company needs to check whether there is any dependency between the
features “Region” and “Manager”.
Help the company to carry out their study with the help of data provided.

# To clarify whether there is any increase in sales after stepping into digital marketing

In [7]:
data['Sales_before_digital_add(in $)'].mean()


149239.95454545456

In [8]:
data['Sales_After_digital_add(in $)'].mean()


231123.72727272726

So here we have found out the mean for the data before digital marketing amd mean for the data after digital marketing..Mean before digital marketing is found to be 149239.95454545456 and mean after digital markrting is found to be 231123.72727272726.

In [33]:


t_stat,p_value=stat.ttest_rel(data['Sales_After_digital_add(in $)'],data['Sales_before_digital_add(in $)'],alternative='greater')
print('T statistic value=',t_stat)
print('p value=',p_value)


T statistic value= 12.09070525287017
p value= 3.168333502287889e-11


In [34]:

if p_value<=0.05:
    print("Reject H0,sales didn't increase")
else:
     print("Accept H0,sales has increased")
 

Reject H0,sales didn't increase


Here H0 is null hypothesis and H1 is alternative hypothesis.
H0(Null Hypothesis):Mean Sales after stepping into digital marketing > Mean Sales before stepping into digital marketing.
H1(Alternate Hypothesis):Mean Sales after stepping into digital marketing <= Mean Sales before stepping into digital marketing.

So based on this it is found that there is no increase in the sales after stepping into digital marketing.

# The company needs to check whether there is any dependency between the features “Region” and “Manager”.

In [19]:
data2=pd.crosstab(data['Manager'],data['Region'])
print(data2)

Region       Region - A  Region - B  Region - C
Manager                                        
Manager - A           4           4           1
Manager - B           3           1           3
Manager - C           3           2           1


In [21]:
val=stat.chi2_contingency(data2)

In [22]:
expected=val[3]

In [23]:
expected

array([[4.09090909, 2.86363636, 2.04545455],
       [3.18181818, 2.22727273, 1.59090909],
       [2.72727273, 1.90909091, 1.36363636]])

In [25]:
no_of_row=len(data2.iloc[0:2,0])
no_of_col=len(data2.iloc[0,0:2])
dof=(no_of_row-1)*(no_of_col-1)
print('Degree Of Freedom:',dof)

Degree Of Freedom: 1


In [26]:
from scipy.stats import chi2
chi_sq=sum([(o-e)**2./e for o,e in zip(observed,expected)])
chi_stat=chi_sq[0]+chi_sq[1]
print('Chi-Square Statistic=',chi_stat)

Chi-Square Statistic= 1.1712018140589566


In [28]:
critical_value=chi2.ppf(q=1-0.05,df=dof)
print('Critical Value=',critical_value)

Critical Value= 3.841458820694124


In [29]:
p_value=chi2.cdf(x=chi_stat,df=dof)
print('P Value=',p_value)
print('Significance Level',0.05)
print('Degree of Freedom',dof)

P Value= 0.720845562353432
Significance Level 0.05
Degree of Freedom 1


In [30]:
if p_value<=0.05:
    print('Reject H0,both are independent')
else:
     print('Accept H0,both are dependent')

Accept H0,both are dependent


 Here,H0: 'Region' and 'Manager' are independent
 H1: 'Region' and 'Manager' are dependent 
 
 so both variables region and manager are dependent